packages/editor/src/components/post-pending-status/check.js

```diff
@@ -31,11 +26,7 @@ export default compose(
 		const { isCurrentPostPublished, getCurrentPostType, getCurrentPost } =
 			select( editorStore );
 		return {
-			hasPublishAction: get(
-				getCurrentPost(),
-				[ '_links', 'wp:action-publish' ],
-				false
-			),
+			hasPublishAction: getCurrentPost()._links?.[ 'wp:action-publish' ],
```

Comment:
`getCurrentPost()` will always be an object:

https://github.com/WordPress/gutenberg/blob/4944315cac156c78313121657dff9d58ce8f4d3f/packages/editor/src/store/selectors.js#L170